snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/validation/SnomedQueryValidationRuleEvaluator.java

```diff
@@ -155,11 +143,14 @@ public final SB prepareSearch() {
 
 		@OverridingMethodsMustInvokeSuper
 		protected SB prepareSearch(SB req) {
-			return req.filterByActive(active)
-					.filterByReleased(released)
-					.filterByModule(module)
-					.filterByEffectiveTime(effectiveTime);
+			req.filterByActive(active)
+			.filterByReleased(released)
+			.filterByEffectiveTime(effectiveTime);
 			
+			if (!filterByWorkingModule) {
```

Comment:
Change this condition according to the cases mentioned above.

Reply:
Will move the logic to `eval()`.